app/controllers/api/v1/study_files_controller.rb

```diff
@@ -300,6 +300,12 @@ def perform_update(study_file)
           end
         end
 
+        # Somewhere in Rails automagic, the upload filename is set as the study file name
+        # We want to have it be the upload_file_name sent by the frontend
+        if safe_file_params[:upload] && safe_file_params[:upload_file_name]
+          safe_file_params[:upload].original_filename = safe_file_params[:upload_file_name]
```

Comment:
I wish we didn't have to do this, but I agree this is a classic case of Rails completely obfuscating functionality and making it impossible to know what's actually going on.